streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -325,6 +328,11 @@ private void recordSendError(final String topic, final Exception exception, fina
         log.error(errorMessage, exception);
     }
 
+    private boolean isMissingMetadata (final Exception exception) {
```

Comment:
Should we rewrite this to:
```
    private boolean isRetriable (final Exception exception) {
        return  !(exception instanceOf TimeoutException && exception.getCause() != null
                && exception.getCause() instanceof UnknownTopicOrPartitionException()) || exception instanceOf RetriableExcepiton;
    }
```

Hope I got the logic right... Basically we want to say, it is retriable if it is a `RetryableExcetpion` but not if it is a TimeoutException with root-cause "unknonw topic or partition".